candle-core/src/pickle.rs

```diff
@@ -757,7 +758,20 @@ impl PthTensors {
             tensor_info.dtype,
             &mut reader,
         )?;
-        Ok(Some(tensor))
+
+        if is_fortran_contiguous {
```

Comment:
Let's maybe add to the condition that there are multiple dimensions (or that it's not c contiguous). Otherwise for array with a single dimension we will pass through this case, the output will be correct but seems a bit unnecessary.